de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/dvlm/mat/MatImporterTest.java

```diff
@@ -0,0 +1,343 @@
+		StructuralElementInstance sei3 = tsei3.getStructuralElementInstance();
+		sei3.setName("testseiChild");
+		TestStructuralElementOther tsei4 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei4 = tsei4.getStructuralElementInstance();
+		sei4.setName("testseiChild2");
+		sei2.setParent(sei3);
+		sei4.setParent(sei);
+		assertFalse("Not the same children", importer.checkIfCorrectSei(sei, mat));
+	}
+
+	@Test
+	public void testImportCasDeleteCas() throws IOException {
+		TestCategoryAllProperty tc2 = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc2);
+		mat = exporter.exportSei(sei);
+		TestCategoryComposition tc1 = new TestCategoryComposition(testConcept);
+		tsei.add(tc1);
+		assertTrue("Sei has two CategoryAssinments", sei.getCategoryAssignments().size() == 2);
+		importer.importSei(sei, mat);
+		assertTrue("Only one CategoryAssinment", sei.getCategoryAssignments().size() == 1);
+		assertTrue("Right CategoryAssinment included",
+				sei.getCategoryAssignments().get(0).getName().equals(tc2.getName()));
+	}
+
+	@Test
+	public void testImportAPIDeleteAPI() throws IOException {
+		EReferenceTest tc2 = new EReferenceTest(testConcept);
+		tsei.add(tc2);
+		TestCategoryAllProperty tc3 = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc3);
+		APropertyInstance nInstance = sei.getCategoryAssignments().get(0).getPropertyInstances().get(0);
+		sei.getCategoryAssignments().remove(0);
+		mat = exporter.exportSei(sei);
+		sei.getCategoryAssignments().get(0).getPropertyInstances().add(nInstance);
+		assertEquals("CategoryAssinment has seven PropertyInstances", sei.getCategoryAssignments().get(0).getPropertyInstances().size(), NUMBEROFELEMENTS + 1);
+		importer.importSei(sei, mat);
+		assertTrue("Instance deleted", sei.getCategoryAssignments().get(0).getPropertyInstances().size() == NUMBEROFELEMENTS);
+	}
+
+	@Test
+	public void testImportOfValuesRemoveAll() throws IOException {
+		TestCategoryAllProperty tc = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc);
+		mat = exporter.exportSei(sei);
+		
+		TestStructuralElement tsei2 = new TestStructuralElement(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		TestCategoryAllProperty tc1 = new TestCategoryAllProperty(testConcept);
+		tc1.setTestBool(false);
+		tc1.setTestFloat(2);
+		tc1.setTestEnum("HIGH");
+		tc1.setTestString("test");
+		tc1.setTestInt(1);
+		URI testUri =  URI.createPlatformResourceURI("Testresource", false);
+		tc1.setTestResource(testUri);
+		sei2.setName("testsei");
+		sei2.setUuid(sei.getUuid());
+		tsei2.add(tc1);
+		importer.importSei(sei2, mat);
+
+		EList<APropertyInstance> caSei = sei.getCategoryAssignments().get(0).getPropertyInstances();
+		EList<APropertyInstance> caSei2 = sei2.getCategoryAssignments().get(0).getPropertyInstances();
+		assertTrue("same number of elements", caSei.size() == caSei2.size());
+		assertEquals("same testString", tc1.getTestString(), tc.getTestString());

```

Comment:
You mixed up expected and actual here, didn't you? You expect that all values of tc1 are overwritten by the import of tc? Then it the order in the assert statement should be the other way around... This way it is confusing because you think that tc should have change to the values you set to tc1.... And please also add a comment here or a more meaningful name of the test case

Reply:
I changed the order and added some comments.